mempool/src/mempool.rs

```diff
@@ -389,6 +390,17 @@ impl Mempool {
                                 .filter(|hash| {
                                     let old_tx = mempool_state.get(hash).unwrap();
 
+                                    // The sender must be able to at least pay the fee (in case the tx fails)
+                                    // (Assuming that all pending txns in the mempool for this sender are included in a block)
+                                    if !sender_account.can_fee_be_paid(
+                                        old_tx,
+                                        new_total + old_tx.fee,
+                                        blockchain.timestamp(),
+                                    ) {
+                                        // If the fee cannot be paid for this transaction, we filter it
+                                        return true;
+                                    }
+
```

Comment:
That is true, the _total_value_ check is stronger than the fee check because _total_value_ already incudes the fee, which seems to be sufficient for basic transactions.
But is this sufficient for other transaction types? i.e: vesting and htlc contracts?